src/panoptes/pocs/focuser/astromechanics.py

```diff
@@ -58,8 +79,18 @@ def connect(self, port):
 
         self._connect(port, baudrate=38400)
 
-        # Return string that makes it clear there is no serial number
-        return "no_serial_number_available"
+        # Send get position command to see if response ends with '#', proper of the astromechs
+        res = self._send_command("P#")
+
+        if res and res.endswith("#"):
+            try:
+                dev = usb.core.find(idVendor=0x0403, idProduct=0x6001)
+                self._serial_number = usb.util.get_string(dev, dev.iSerialNumber)
+                # Return string that makes it clear there is no serial number
+                return self._serial_number
+
+            except Exception:
+                raise error.PanError(f"{self.name} not found in {self.port}")
```

Comment:
I still think the exception should be raised when the `if` fails, not when the serial number check fails.  And can we check it for a valid position response rather than the generic `#`?